tensorflow_probability/python/experimental/bijectors/highway_flow_test.py

```diff
@@ -0,0 +1,127 @@
+            elif activation == 'tanh':
+                activation_fn = tf.nn.tanh
+            elif activation == 'none':
+                activation_fn = None
+
+            bijector = tfp.experimental.bijectors.HighwayFlow(
+                width=width,
+                residual_fraction=residual_fraction,
+                activation_fn=activation_fn,
+                bias=tf.Variable(0., dtype=dtype),
+                upper_diagonal_weights_matrix=tf.Variable(tf.eye(width)),
+                lower_diagonal_weights_matrix=tf.Variable(tf.eye(width))
+            )
+
+            self.evaluate([v.initializer for v in bijector.trainable_variables])
+            x = tf.ones((batch_size, width)) # * samplers.uniform((batch_size, width), -10., 10., seed=seed)
+            if activation == 'none':
+                y = x
+            else:
+                y = residual_fraction * x + (1 - residual_fraction) * activation_fn(x)
+            expected_forward_log_det_jacobian = _activation_log_det_jacobian(x, residual_fraction, activation)
+            expected_inverse_log_det_jacobian = -expected_forward_log_det_jacobian
+            self.assertAllClose(y, bijector.forward(x))
+            self.assertAllClose(x, bijector.inverse(y))
+            self.assertAllClose(
+                expected_inverse_log_det_jacobian,
+                bijector.inverse_log_det_jacobian(y, event_ndims=1),
+            )
+            self.assertAllClose(
+                expected_forward_log_det_jacobian,
+                bijector.forward_log_det_jacobian(x, event_ndims=1),
+            )
+
+
+    def testResidualFractionGradientsWithCenteredDifference(self):
+        width = 4
+        batch_size = 3
+        dtype = tf.float32
+        residual_fraction = tf.Variable(0.5)
+        bijector = tfp.experimental.bijectors.HighwayFlow(
+            width=width,
+            residual_fraction=residual_fraction,
+            activation_fn=tf.nn.softplus,
+            bias=tf.Variable(0., dtype=dtype),
+            upper_diagonal_weights_matrix=tf.Variable(tf.eye(width)),
+            lower_diagonal_weights_matrix=tf.Variable(tf.eye(width))
+        )
+        target = tfd.MultivariateNormalDiag(loc=tf.zeros(width), scale_diag=tf.ones(width))
+        x = tf.ones((batch_size, width))
+        with tf.GradientTape() as g:
+            g.watch(bijector.residual_fraction)
+            y = tf.reduce_mean(target.log_prob(bijector.forward(x)))
+        tf_grad = g.gradient(y, bijector.residual_fraction)
+
+        h = 1e-4
+        eps = 1e-6
+
+        bijector.residual_fraction = residual_fraction + h
+        y1 = tf.reduce_mean(target.log_prob(bijector.forward(x)))
+        bijector.residual_fraction = residual_fraction - h
+        y2 = tf.reduce_mean(target.log_prob(bijector.forward(x)))
+
+        manual_grad = (y1 - y2) / (2 * h)
+
+        diff = tf.math.abs(tf_grad - manual_grad) / tf.reduce_max((eps, tf.math.abs(tf_grad) + tf.math.abs(manual_grad)))
```

Comment:
It's probably clearer (and easier) to write just `self.assertAllClose(tf_grad, manual_grad, rtol=1e-4)` in place of this explicit calculation.

Reply:
I have changed this but then to pass the test I needed to increase h to 1e-3. Could you please double-check that the gradient test I implemented is reliable?